src/fabric_cicd/publish.py

```diff
@@ -116,106 +91,97 @@ def publish_all_items(fabric_workspace_obj: FabricWorkspace, item_name_exclude_r
-        collect_published_items_for_type("Eventhouse")
     if "KQLDatabase" in fabric_workspace_obj.item_type_in_scope:
         print_header("Publishing KQL Databases")
         items.publish_kqldatabases(fabric_workspace_obj)
-        collect_published_items_for_type("KQLDatabase")
     if "KQLQueryset" in fabric_workspace_obj.item_type_in_scope:
         print_header("Publishing KQL Querysets")
         items.publish_kqlquerysets(fabric_workspace_obj)
-        collect_published_items_for_type("KQLQueryset")
     if "Reflex" in fabric_workspace_obj.item_type_in_scope:
         print_header("Publishing Activators")
         items.publish_activators(fabric_workspace_obj)
-        collect_published_items_for_type("Reflex")
     if "Eventstream" in fabric_workspace_obj.item_type_in_scope:
         print_header("Publishing Eventstreams")
         items.publish_eventstreams(fabric_workspace_obj)
-        collect_published_items_for_type("Eventstream")
     if "KQLDashboard" in fabric_workspace_obj.item_type_in_scope:
         print_header("Publishing KQL Dashboards")
         items.publish_kqldashboard(fabric_workspace_obj)
-        collect_published_items_for_type("KQLDashboard")
     if "Dataflow" in fabric_workspace_obj.item_type_in_scope:
         print_header("Publishing Dataflows")
         items.publish_dataflows(fabric_workspace_obj)
-        collect_published_items_for_type("Dataflow")
     if "DataPipeline" in fabric_workspace_obj.item_type_in_scope:
         print_header("Publishing Data Pipelines")
         items.publish_datapipelines(fabric_workspace_obj)
-        collect_published_items_for_type("DataPipeline")
     if "GraphQLApi" in fabric_workspace_obj.item_type_in_scope:
         print_header("Publishing GraphQL APIs")
         logger.warning(
             "Only user authentication is supported for GraphQL API items sourced from SQL Analytics Endpoint"
         )
         items.publish_graphqlapis(fabric_workspace_obj)
-        collect_published_items_for_type("GraphQLApi")
 
     # Check Environment Publish
     if "Environment" in fabric_workspace_obj.item_type_in_scope:
         print_header("Checking Environment Publish State")
         items.check_environment_publish_state(fabric_workspace_obj)
     
+    # Refresh deployed items after publishing to get the updated state
+    fabric_workspace_obj._refresh_deployed_items()
+    
+    # Collect published items information using deployed_items and repository_items
+    published_items = {}
+    for item_type in fabric_workspace_obj.item_type_in_scope:
+        if item_type in fabric_workspace_obj.repository_items:
+            published_items[item_type] = {}
+            for item_name in fabric_workspace_obj.repository_items[item_type]:
+                # Get the deployed item info (contains the actual GUID from workspace)
+                deployed_item = fabric_workspace_obj.deployed_items.get(item_type, {}).get(item_name)
```

Comment:
use self.workspace_items instead so we can obtain the sqlendpoint (if no value default to "")